deltachat-ios/Controller/ChatListController.swift

```diff
@@ -151,12 +151,8 @@ class ChatListController: UITableViewController {
         if let msgsNoticedObserver = self.msgsNoticedObserver {
             nc.removeObserver(msgsNoticedObserver)
         }
-        if let foregroundObserver = self.foregroundObserver {
-            nc.removeObserver(foregroundObserver)
-        }
-        if let backgroundObserver = self.backgroundObserver {
-            nc.removeObserver(backgroundObserver)
-        }
+        // remove non-block observers
+        nc.removeObserver(self)
```

Comment:
we can do that in this case, but need to be careful, if not all observers are added and removed at the same step of the lifecycle. The documentation recommends therefore: 

> When removing an observer, remove it with the most specific detail possible. For example, if you used a name and object to register the observer, use removeObserver(_:name:object:) with the name and object.